assets/src/blocks/Spreadsheet/SpreadsheetEditor.js

```diff
@@ -0,0 +1,148 @@
+  },
+  // Blue variables
+  '#c9e7fa': {
+    'spreadsheet-header-background': '#074365',
+    'spreadsheet-even-row-background': '#e7f5fe',
+    'spreadsheet-odd-row-background': '#c9e7fa'
+  }
+};
+
+export class SpreadsheetEditor extends Component {
+  constructor(props) {
+    super(props);
+    this.handleErrors = this.handleErrors.bind(this);
+    this.state = {
+      invalidSheetId: false,
+      errorFetchingSpreadsheet: false,
+    };
+  }
+
+  handleErrors( errors ) {
+    console.log(errors);
+    this.setState( errors );
+  }
+
+  renderEdit() {
+    const {__} = wp.i18n;
+
+    const { attributes, setAttributes } = this.props;
+
+    const toAttribute = attributeName => value => {
+      setAttributes( { [ attributeName ]: value } );
+    };
+
+    const toCssVariables = ( value ) => {
+      setAttributes( {
+        css_variables: colors_variables_map[value]
+      } );
+    };
+
+    return (
+      <Fragment>
+        <InspectorControls>
+          <PanelBody title={__('Setting', 'p4ge')}>
+            <ColorPaletteControl
+              label={__('Table Color', 'p4ge')}
+              value={ attributes.css_variables['spreadsheet-odd-row-background'] }
+              onChange={ toCssVariables }
+              disableCustomColors
+              clearable={ false }
+              options= { colors }
+            />
+            <TextControl
+              label={__('Spreadsheet URL', 'planet4-blocks-backend')}
+              placeholder={__('Enter Google Spreadsheet URL', 'planet4-blocks-backend')}
+              value={ attributes.url }
+              onChange={ toAttribute( 'url' ) }
+            />
+            <div className="sidebar-blocks-help">
+              <ul>
+                <li>
+                  { __(`From Your Google Spreadsheet Table choose File -> Publish on web.
+                  No need to choose the output format, any of them will work.
+                  A pop-up window will show up, click on the Publish button and then OK when the confirmation message is displayed.
+                  Copy the URL that is highlighted and paste it in this block.`, 'planet4-blocks-backend') }
+                </li>
+                <li>
+                  { __(`If you make changes to the sheet after publishing
+                    then these changes do not always immediately get reflected,
+                    even when "Automatically republish when changes are made" is checked.`, 'planet4-blocks-backend') }
+                </li>
+                <li>
+                  { __(`You can force an update by unpublishing and republishing the sheet.
+                    This will not change the sheet's public url.`, 'planet4-blocks-backend' ) }
+                </li>
+              </ul>
+            </div>
+          </PanelBody>
+        </InspectorControls>
+      </Fragment>
+    );
+  }
+
+  renderView() {
+    const {__} = wp.i18n;
+
+    return <Fragment>
+      {
+        ! this.props.attributes.url
```

Comment:
`this.props.attributes` is used a few times here, maybe easier on the eye if we destructure it `const { attributes } = this.props`.